HW0/P5/P5.txt

```diff
@@ -1 +1,10 @@
-# Your answers here.
+1) Adding together the numbers takes 255 operations (seconds) since each number has to be added in at some point
+2) It would take 34 seconds. You give each friend 32 numbers to add which takes 31 seconds and produces 8 sums. Then you pair wise add to 4 sums, once again to 2, and finally one more second to get to one. This is a total of 34 seconds. 
+3) If everyone is working it is the log base 2 of 256 which is 8 seconds. This only requires 128 people, any more would be redundant
+4) Like part 3 it's the log base two of N since you add two numbers at a time and combine them and repeat. On the plot the blue is the linear time it takes one person to add the numbers, and the green is the logarithmic time 
+5) This is 15 seconds. It takes twice as long as in part three because after every round of counting, you need exactly one second for half the people to tell the other half. It is 1 less than double because at the end you don't need to tell anyone the final count. 
+6) Assuming it still takes one second to communicate a number, the answer becomes 512. Starting with the simple scenario of two bags, after two seconds you have handed out both bags, third second to communicate a number, fourth second to add. With a third bag, the person has to communicate the number, and an addition has to be made again giving six seconds. The addition of a bag leads to an additional two seconds in every case. 
+
+If you give out two bags at a time it takes 257 seconds. That is because each additional two bags you give out requires the person to add them, and tell it to another person while the next bags are being given out and added. Four bags takes five seconds, at the end of 2s person two has the bags, at the end of 3s person two has added, at the end of 4s person two tells person one, at the end of 5 person one has added together. Then you go from there and the answer becomes 257. I'm assuming each person can only do one thing at a time, for example you can't give a bag while recieving a number at the same time. 
+
+The fastest way to do it is as follows. Hand half to one person. Then you and that person hand half to two others and so on until everyone has 2 bags after 7 seconds. Then everyone adds those two, and half tell someone else. That continues until there is one sum after 20 seconds.
```

Comment:
Since you're not a cashier and are presumably unwilling to help the cashiers count your pennies, the first step should be you handing all the bags to one cashier, and then the rest progresses like you said (this only adds one second, though).